### DISCUSSION TITLE
API for Creating and Setting View Area [AXIS P1375 S/N]

### DISCUSSION THREAD

**Discussion by paxygithub** [AxisCommunications/discussions#105] [VAPIX] [answered]
I am developing an application using a P1375 and P1375-E camera. I want to create and set a view area from the application via HTTP API requests. I tried configuring an already existing view area using a POST request to http://192.168.150.13/axis-cgi/viewarea/configure.cgi with the following JSON body:

{
"apiVersion": "1.0",
"context": "123",
"method": "setGeometry",
"params": {
"viewArea": {
"id": 1000002,
"rectangularGeometry": {
"horizontalOffset": 0,
"horizontalSize": 960,
"verticalOffset": 0,
"verticalSize": 540
}
}
}

The server responds with 404 Not Found. Does this endpoint exist for this brand of camera? If so, why isn't the API call working for me?

**Comment by shadysp** [comment]
Tested  the same example on P1375 Firmware:10.12 and it works. 
What response do you get when you list for view areas.

//ip-add/axis-cgi/viewarea/info.cgi

JSON input parameters
{
  "apiVersion": "1.0",
  "context": "123",
  "method": "list"
}

Can you also try to retrieve information about the APIs installed on this device.
https://www.axis.com/vapix-library/subjects/t10175981/section/t10131704/display

Can you test with curl.

**Reply by paxygithub** [reply]
Hey, sorry for the late response!
I'm using Firmware version 9.80.2.4 with the P1375 model.

When using the address **//ip-add/axis-cgi/viewarea/info.cgi** with the JSON body using curl, I get the following response:

<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL was not found on this server.</p>
</body></html>

**Not Found
The requested URL was not found on this server**

If I use the following curl command:

**curl -i -X POST -H "Content-Type: application/json" --anyauth -uroot:pass http://192.168.150.13/axis-cgi/apidiscovery.cgi --data "{ \"method\": \"getSupportedVersions\" }"**

I get the following:

**HTTP/1.1 200 OK
Date: Thu, 03 Nov 2022 20:49:28 GMT
Server: Apache/2.4.46 (Unix) OpenSSL/1.1.1g
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
Vary: Accept-Encoding
Transfer-Encoding: chunked
Content-Type: application/json;charset=utf8

{"method": "getSupportedVersions", "data": {"apiVersions": ["1.0"]}}**

**Comment by paxygithub** [comment, marked as answer]
Just upgraded the firmware to 11.0 and now the camera responds with the view area API. Noticed you were using a higher FW version. Thank you for all the help! Really appreciate it shadysp
